src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -2265,29 +2286,56 @@ bool Session::client_reset_if_needed()
         return false;
     }
 
+    // Save a copy of the status and action in case an error/exception occurs
+    Status cr_status = client_reset_config->error;
+    ProtocolErrorInfo::Action cr_action = client_reset_config->action;
+
     auto on_flx_version_complete = [this](int64_t version) {
         this->on_flx_sync_version_complete(version);
     };
-    bool did_reset =
-        client_reset::perform_client_reset(logger, *get_db(), std::move(*client_reset_config), m_client_file_ident,
-                                           get_flx_subscription_store(), on_flx_version_complete);
-
-    call_debug_hook(SyncClientHookEvent::ClientResetMergeComplete);
-    if (!did_reset) {
+    try {
+        // Storage for the file ident from the fresh realm that was migrated to the local realm
+        SaltedFileIdent client_file_ident_out;
```

Comment:
I think it would be better to reread the ident from the file rather than using an out parameter. This would make it harder to accidentally have inconsistent states where we fail to actually update the ident in the file and don't notice.

Reply:
Updated - the `file_ident_out` parameter has been removed and the local realm file ident is being extracted after the client reset diff is complete.